addusers.py

```diff
@@ -421,6 +449,7 @@ def parse_rows(rows, select_user=None):
                     if ks_user:
                         raise ItemExistsError('User', user.name)
                     new_ks_user = openstack.create_user(user, ks_project)
+                    user_dict[new_ks_user.name] = new_ks_user.id
```

Comment:
Do you actually need the extra dictionary here?  `user.row` preserves the row index of the original rows data structure.  You could add the info you want to the correct row here, and avoid storing the info and looping through the rows again later.